server/internal/adapter/publicapi/types.go

```diff
@@ -65,10 +65,15 @@
 	return json.Marshal(m)
 }
 
-func NewItem(i *item.Item, s *schema.Schema, assets asset.List, urlResolver asset.URLResolver) Item {
+func NewItem(i *item.Item, s *schema.Schema, groupSchemas schema.List, assets asset.List, urlResolver asset.URLResolver) Item {
```

Comment:
its better to refactor it to use schema package

Reply:
I know that the function can be prettier, but as the group ID is not involved in the implementation of `NewItem` I think calculating it will just add extra work